### PR TITLE
Scheduler rework

### PR DESCRIPTION
**Pull request by Ugon**
I've implemented the changes we were talking about during last Lambda Academy meeting. You can see them working by executing this code:

``` erlang
f().
Filename = "talks.txt".

StartAtS = calendar:datetime_to_gregorian_seconds(calendar:local_time()) + 5.
StartAt = calendar:gregorian_seconds_to_datetime(StartAtS).
EndAt = calendar:gregorian_seconds_to_datetime(StartAtS + 120).

StartGenAtS = StartAtS + 10.
StartGenAt = calendar:gregorian_seconds_to_datetime(StartGenAtS).

tt_gen:gen_talks(StartGenAt, 30, 5, Filename).
tt_importer:import_file(Filename).
tt_scheduler:schedule(StartAt, EndAt, {0, 0, 3}).
```

I tried to implement it so that each talk can be published at most once (unless 'published marker' in tt_store is explicitly erased), so that when they are moved last minute, confusing tweets wont be generated. I assumed that when a talk is moved, a custom tweet will be sent directly via Twitter.

I encountered a problem with tt_scheduler tests. They give me an error:

```
** Reason for termination == 
** {'function not exported',
       [{tt_store,find_by_time_unpublished,
            [{{2016,2,3},{20,13,49}},{{2016,2,3},{20,13,50}}],
            []},
        {tt_scheduler,do_action,2,
            [{file,
                 "/home/ugon/development/projects/erlang/2015.12_erlang/_build/test/lib/talks_tweeter/src/tt_scheduler.erl"},
             {line,105}]},
        {tt_scheduler,handle_cast,2,
            [{file,
                 "/home/ugon/development/projects/erlang/2015.12_erlang/_build/test/lib/talks_tweeter/src/tt_scheduler.erl"},
             {line,68}]},
        {gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,615}]},
        {gen_server,handle_msg,5,[{file,"gen_server.erl"},{line,681}]},
        {proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]}
```

The function in question (tt_store:find_by_time_unpublished) is exported, so idk what else could be causing this.

Also, since tt_scheduler depends heavily on time, I don't know how to efficiently test it without writing tests that take at least a couple of seconds to execute.

### CODE REVIEW

File: src/talks_tweeter_sup.erl [hunk 1]
```diff
@@ -48,8 +48,11 @@ init([]) ->
                        type => worker,
                        modules => [tt_importer]
                       },
+
+    ActionInterval = {0, 0, 1},
+    TimeWindow = {0, 0, 3},
     SchedulerWorker = #{id => tt_scheduler,
-                        start => {tt_scheduler, start_link, []},
+                        start => {tt_scheduler, start_link, [ActionInterval, TimeWindow]},
```
Comment (Ugon) [inline src/talks_tweeter_sup.erl:55]: ActionInterval is time between tt_scheduler asking tt_store for talks between (TimeBeforeTalk; TimeBeforeTalk + TimeWindow).

**Comment by mentels** [comment]
That is really great!!! Thanks!!!

I pushed the fixes so that the tests pass into the scheduler-rework branch. The thing with `function not exported` was about the mock of `tt_store`. When mocking with `meck` you have to provide expectations about function calls.

Look into the code for the rest of the comments.

**Comment by mentels** [comment]
Regarding the testing: there could be some kind of a switch, that would convert times few magnitudes down. So that when an environment variable is set, the seconds are interpreted as milliseconds... but that requires some work.

File: .idea/modules.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mentels) [inline .idea/modules.xml]: This file should not be in the repo

File: .idea/vcs.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (mentels) [inline .idea/vcs.xml]: This file should no be in the repo

File: 2015.12_erlang.iml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="ERLANG_MODULE" version="4">
```
Comment (mentels) [inline 2015.12_erlang.iml]: This file should not be in the repo

File: src/talks_tweeter_sup.erl (hunk 1 above)
Comment (mentels) [inline src/talks_tweeter_sup.erl:55]: This options, `ActionInterval` and `TimeWindows` should be put into the [config/sys.config](https://github.com/lambdaacademy/2015.12_erlang/blob/master/config/sys.config) file as following:

``` erlang
[
 {talks_tweeter, [{action_interval, {0,0,1},
                          {time_window, {0,0,3}]},           
{etweet, [
             ....
             ]
 }].
```

Then, this file is read upon the start up (see [here](https://github.com/lambdaacademy/2015.12_erlang/blob/master/Makefile#L35) and the environment variables can be accessed via `application:get_env/3`, for example:

``` erlang
ActionInterval = application:get_env(_AppName = talks_tweeter, _Key = action_interval, _Default = {0,0,1}),
TimeWindow = application:get_env(talks_tweeter, time_window, {0,0,3}).
```

So let's do it the right way.

**Pull request closed without merging**